rocketpy/mathutils/function.py

```diff
@@ -2197,10 +2203,53 @@ def __sub__(self, other):
         result : Function
             A Function object which gives the result of self(x)-other(x).
         """
-        try:
-            return self + (-other)
-        except TypeError:
-            return Function(lambda x: (self.get_value_opt(x) - other(x)))
+        self_array = isinstance(self.source, np.ndarray)
+        other_is_func = isinstance(other, Function)
+        other_array = isinstance(other.source, np.ndarray) if other_is_func else False
+        inputs = self.__inputs__[:]
+        interp = self.__interpolation__
+        extrap = self.__extrapolation__
+        dom_dim = self.__dom_dim__
+
+        if self_array and other_array and np.array_equal(self._domain, other._domain):
+            source = np.column_stack((self._domain, self._image - other._image))
+            outputs = f"({self.__outputs__[0]}-{other.__outputs__[0]})"
+            return Function(source, inputs, outputs, interp, extrap)
+        elif isinstance(other, NUMERICAL_TYPES) or self.__is_single_element_array(
+            other
+        ):
+            if self_array:
+                source = np.column_stack(
+                    (self._domain, np.subtract(self._image, other))
+                )
+                outputs = f"({self.__outputs__[0]}-{other})"
+                return Function(source, inputs, outputs, interp, extrap)
+            else:
+                return Function(
+                    self.__make_arith_lambda(
+                        operator.sub, self.get_value_opt, other, dom_dim
+                    ),
+                    inputs,
+                )
+        elif callable(other):
+            if other_is_func:
+                other_dim = other.__dom_dim__
+                other = other.get_value_opt if other_array else other.source
+            else:
+                other_dim = len(signature(other).parameters)
+
+            if dom_dim == 1 or other_dim == 1 or dom_dim == other_dim:
+                return Function(
+                    self.__make_arith_lambda(
+                        operator.sub, self.get_value_opt, other, dom_dim, other_dim
+                    )
+                )
+            else:
+                raise TypeError(
+                    f"The number of parameters in the function to be subtracted ({other_dim}) "
+                    f"does not match the number of parameters of the Function ({dom_dim})."
+                )
+        raise TypeError("Unsupported type for subtraction")
```

Comment:
Are all these operation methods the same (add, sub, mul...)? They seem really similar if not equal, except for the operation itself. Can we put all these checks and Function creations in a separate function? And then just call this inside each of the operations methods?

Reply:
Okay, let it go for now then. We can get back to this if it is ever needed